### DISCUSSION TITLE
Sharing datasets cache across different users in linux

### DISCUSSION THREAD

**Discussion by halean** [huggingface/datasets#2467] [Q&A]
Is it possible to share the datasets cache (and transformer models cache) to different users in ubuntu. I try to use the same cache location, but got the permission errors for the lock file.
Many thanks.

**Comment by mariosasko** [comment]
Hi! You can easily share the dataset by pushing it to the Hub (`ds.push_to_hub`), or if you want to keep the dataset private, you can create an organization on the Hub, invite other members and then push the dataset as follows:
```python
# push to the Hub
ds.push_to_hub(f"{org_name}/{ds_name}", private=True)
# load from the Hub
ds = load_dataset(f"{org_name}/{ds_name}", use_auth_token=True)
```

**Reply by kleag** [reply]
Hi, I have the same question and your answer is not a solution for me because the reason we have to share a dataset is that we are on a machine that don't have access to the Internet when running our models. So, we want to download and cache the dataset one time, in a read only mode, and use it for different users. This is for public datasets on huggingface.

**Comment by FarisHijazi** [comment]
have you tried creating symbolic links using `ln -s otheruser/.cache ~/.cache`?

**Comment by AlanBlanchet** [comment]
You can maybe try changing the ENV variables for the cache to some shared space on a disk or private network.

Personnaly I have this piece of code in `/etc/profile.d/custom_env.sh` :
```bash
# Users can share hugging face datasets
HF="/disk/huggingface"
export HF_DATASETS_CACHE="$HF/datasets"
export HF_HUB_CACHE="$HF"
```
